test/integration/main_test.go

```diff
@@ -358,6 +355,155 @@ deletion_timeline:
 	}
 }
 
+// TestMappingReusableWorkflowCall tests the mapping file-copy reusable workflow has successfully
+// uploaded the file to gcs and triggered pmap event handler, and write the corresponding entry
+// into bigquery.
+func TestMappingReusableWorkflowCall(t *testing.T) {
+	t.Parallel()
+	cases := []struct {
+		name                string
+		bigqueryTable       string
+		traceID             string
+		wantResourceMapping *v1alpha1.ResourceMapping
+	}{
+		{
+			name:          "mapping-copy-success-event",
+			bigqueryTable: cfg.MappingTableID,
+			traceID:       "fake-pmap-dev-manual-test-mapping",
+			wantResourceMapping: &v1alpha1.ResourceMapping{
+				Resource: &v1alpha1.Resource{
+					Provider: "gcp",
+					Name:     fmt.Sprintf("//storage.googleapis.com/%s", "pmap-static-ci-bucket-7faa"),
+				},
+				Contacts: &v1alpha1.Contacts{Email: []string{"pmap.mapping@gmail.com"}},
+			},
+		},
+	}
+
+	for _, tc := range cases {
+		tc := tc
+
+		t.Run(tc.name, func(t *testing.T) {
+			t.Parallel()
+
+			ctx := context.Background()
+
+			t.Logf("using trace ID %s", tc.traceID)
+			t.Logf("using workflow run ID %s", cfg.WorkflowRunID)
+
+			// Check if the file uploaded exists in BigQuery.
+			// We use workflowRunId here as the unique identifier as this is the test the
+			// reusable workflow and the traceID is written in the yaml file, which won't
+			// be unique if we don't change the yaml file. But workflowRunID is unique for
+			// all workflow runs, and using workflowRunId to query, we can make sure the
+			// reusable workflow uploaded the correct GCS object metadata, and later we only
+			// need to diff the enties in the yaml file.
+			queryString := fmt.Sprintf("SELECT * FROM `%s.%s.%s`", cfg.ProjectID, cfg.BigQueryDataSetID, tc.bigqueryTable)
+			queryString += `WHERE JSON_VALUE(data.githubSource.workflowRunId) = ?`
+
+			bqQuery := bqClient.Query(queryString)
+			bqQuery.Parameters = []bigquery.QueryParameter{{Value: cfg.WorkflowRunID}}
+
+			gotBQEntry := testGetFirstMatchedBQEntryWithRetries(ctx, t, bqQuery, cfg)
+
+			gotPmapEvent := &v1alpha1.PmapEvent{}
+			if err := protojson.Unmarshal([]byte(gotBQEntry.Data), gotPmapEvent); err != nil {
+				t.Fatalf("failed to unmarshal BQEntry.Data to pmapevent: %v", err)
+			}
+
+			gotResourceMapping := &v1alpha1.ResourceMapping{}
+			if err := gotPmapEvent.GetPayload().UnmarshalTo(gotResourceMapping); err != nil {
+				t.Fatalf("failed to unmarshal to resource mapping: %v", err)
+			}
+
+			cmpOpts := []cmp.Option{
+				protocmp.Transform(),
+				protocmp.IgnoreFields(&v1alpha1.ResourceMapping{}, "annotations"),
+			}
+
+			if diff := cmp.Diff(tc.wantResourceMapping, gotResourceMapping, cmpOpts...); diff != "" {
+				t.Errorf("gotPayload unexpected diff (-want,+got):\n%s", diff)
+			}
+
+			if diff := cmp.Diff(tc.traceID, gotResourceMapping.Annotations.AsMap()["traceID"]); diff != "" {
```

Comment:
Nits: Do you need to compare the trace id here? :

Reply:
Yes, to test the right file is uploaded.